### PR TITLE
Quadchute cleanup and instant back transition

### PR DESCRIPTION
**Pull request by AndreasAntener**
Replaces #5182

### CODE REVIEW

**Comment by sanderux** [comment]
maybe we should rename the function while we're at it

File: src/modules/vtol_att_control/vtol_type.cpp [hunk 1]
```diff
@@ -180,27 +180,25 @@ void VtolType::update_fw_state()
...
 	return !_armed->armed || _land_detected->landed;
 }
+
+void VtolType::check_quadchute_condition()
+{
+	// fixed-wing minimum altitude
+	if (_params->fw_min_alt > FLT_EPSILON && _armed->armed) {
+		if (-(_local_pos->z) < _params->fw_min_alt) {
+			_attc->abort_front_transition("Minimum altitude");
+		}
```
Comment (sanderux) [inline src/modules/vtol_att_control/vtol_type.cpp]: @AndreasAntener 
Should we add !_landed so bench testing remains possible?
A reason not to do so would be false positives on land detector

Comment (AndreasAntener) [inline src/modules/vtol_att_control/vtol_type.cpp]: No objection (if it reports landed while flying you'll have bigger problems, like in-air disarm..). Only I don't know how much it's going to help you as the landed state switches quickly when you move the vehicle.

Comment (sanderux) [inline src/modules/vtol_att_control/vtol_type.cpp]: Maybe a minimum altitude to deploy quadchute?
LPOS.Z < -1.5f ?

Comment (AndreasAntener) [inline src/modules/vtol_att_control/vtol_type.cpp]: I think you'll hit that randomly. Probably best to just disable it if you need to do those test. I can't think of a condition that works well.

Comment (sanderux) [inline src/modules/vtol_att_control/vtol_type.cpp]: -(_local_pos->z) < _params->fw_min_alt && -(_local_pos->z) > (_params->fw_min_alt / 2) 

?

**Comment by sanderux** [comment]
Good to merge from my end, will revisit the bench testing later

**Pull request merged**